percona-packaging/scripts/psmdb_builder.sh

```diff
@@ -0,0 +1,796 @@
+    then
+        echo "SRC RPM will not be created"
+        return;
+    fi
+    if [ "x$OS" = "xdeb" ]
+    then
+        echo "It is not possible to build src rpm here"
+        exit 1
+    fi
+    cd $WORKDIR
+    get_tar "source_tarball"
+    rm -fr rpmbuild
+    ls | grep -v tar.gz | xargs rm -rf
+    TARFILE=$(find . -name 'percona-server-mongodb-*.tar.gz' | sort | tail -n1)
+    SRC_DIR=${TARFILE%.tar.gz}
+    #
+    mkdir -vp rpmbuild/{SOURCES,SPECS,BUILD,SRPMS,RPMS}
+    tar vxzf ${WORKDIR}/${TARFILE} --wildcards '*/percona-packaging' --strip=1
+    SPEC_TMPL=$(find percona-packaging/redhat -name 'percona-server-mongodb.spec.template' | sort | tail -n1)
+    #
+    cp -av percona-packaging/conf/* rpmbuild/SOURCES
+    cp -av percona-packaging/redhat/mongod.* rpmbuild/SOURCES
+    #
+    sed -i 's:@@LOCATION@@:sysconfig:g' rpmbuild/SOURCES/*.service
+    sed -i 's:@@LOCATION@@:sysconfig:g' rpmbuild/SOURCES/percona-server-mongodb-helper.sh
+    sed -i 's:@@LOGDIR@@:mongo:g' rpmbuild/SOURCES/*.default
+    sed -i 's:@@LOGDIR@@:mongo:g' rpmbuild/SOURCES/percona-server-mongodb-helper.sh
+    #
+    sed -e "s:@@SOURCE_TARBALL@@:$(basename ${TARFILE}):g" \
+    -e "s:@@VERSION@@:${VERSION}:g" \
+    -e "s:@@RELEASE@@:${RELEASE}:g" \
+    -e "s:@@SRC_DIR@@:$SRC_DIR:g" \
+    ${SPEC_TMPL} > rpmbuild/SPECS/$(basename ${SPEC_TMPL%.template})
+    mv -fv ${TARFILE} ${WORKDIR}/rpmbuild/SOURCES
+    rpmbuild -bs --define "_topdir ${WORKDIR}/rpmbuild" --define "dist .generic" rpmbuild/SPECS/$(basename ${SPEC_TMPL%.template})
+    mkdir -p ${WORKDIR}/srpm
+    mkdir -p ${CURDIR}/srpm
+    cp rpmbuild/SRPMS/*.src.rpm ${CURDIR}/srpm
+    cp rpmbuild/SRPMS/*.src.rpm ${WORKDIR}/srpm
+    return
+}
+
+build_rpm(){
+    if [ $RPM = 0 ]
+    then
+        echo "RPM will not be created"
+        return;
+    fi
+    if [ "x$OS" = "xdeb" ]
+    then
+        echo "It is not possible to build rpm here"
+        exit 1
+    fi
+    SRC_RPM=$(basename $(find $WORKDIR/srpm -name 'Percona-Server-MongoDB-*.src.rpm' | sort | tail -n1))
+    if [ -z $SRC_RPM ]
+    then
+        SRC_RPM=$(basename $(find $CURDIR/srpm -name 'Percona-Server-MongoDB-*.src.rpm' | sort | tail -n1))
+        if [ -z $SRC_RPM ]
+        then
+            echo "There is no src rpm for build"
+            echo "You can create it using key --build_src_rpm=1"
+            exit 1
+        else
+            cp $CURDIR/srpm/$SRC_RPM $WORKDIR
+        fi
+    else
+        cp $WORKDIR/srpm/$SRC_RPM $WORKDIR
+    fi
+    cd $WORKDIR
+    rm -fr rpmbuild
+    mkdir -vp rpmbuild/{SOURCES,SPECS,BUILD,SRPMS,RPMS}
+    cp $SRC_RPM rpmbuild/SRPMS/
+
+    cd rpmbuild/SRPMS/
+    rpm2cpio ${SRC_RPM} | cpio -id
+    TARF=$(find . -name 'percona-server-mongodb-*.tar.gz' | sort | tail -n1)
+    tar vxzf ${TARF} --wildcards '*/buildscripts' --strip=1
+    if [ "x${RHEL}" == "x6" ]; then
+    pip2.7 install --user -r buildscripts/requirements.txt
+    else
+    pip install --user -r buildscripts/requirements.txt
```

Comment:
3.2 doesn't have this file, not sure if this script should build 3.2 also
also maybe consider moving this into install_deps() function, because currently I see it scattered around in different functions

Reply:
Well as it depends on the appropriate branch I think it isn't good idea to move it to depth. As install-deps flag can be used without branch flag.

Reply:
Ok, I just thought it would be cleaner, but if it doesn't fit the workflow it's fine.